### ISSUE TITLE
Erase JMonet

### ISSUE THREAD

**Issue by Javier-Pazos** [defano/jmonet#41]
Hi Matt!

I'm developing a javafx application which contains a jmonet canvas. The application has several fxml pages and controllers. 
The jMonetCanvas is contained in a javafx controller. When I call this controller, it creates a static jMonetCanvas. But I have a problem:

If I go to another controller, the jMonet object continues staying on memory. I'm trying to delete the jMonetObject, calling the dispose method, setting to null the jMonetObject, but it continues in memory with its bufferedImages, listeners,...

What can I do to erase this object completely?  I want to erase completely this object when I go to another controller, and I want to create a new jMonet object when I go to its controller.

Thanks.

**Comment by defano** [comment]
Hi @Javier-Pazos ,

I think have this working now. I've published a new version (0.4.1) that address this.

There were two issues:

1. I was missing a `.dispose()` method on the `JFXPaintCanvasNode` object.
2. JavaFX is a little weird about cleaning things up. Simply removing the canvas from its parent `Pane` keeps a reference to the canvas (or at least, perhaps, the rendered image that was last painted to the screen). Seems that the parent must be garbage collected before all memory will be released.

See this example:

```
public class NewJFXTester extends Application {

    JFXPaintCanvasNode myCanvasNode;
    Pane pane = new Pane();

    public static long memoryInUseMb() {
        return (Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()) / 1024 / 1024;
    }

    @Override
    public void start(Stage stage) throws InterruptedException {

        System.out.println("Initial memory in use: " + memoryInUseMb());

        // Create a pane for it
        myCanvasNode = new JFXPaintCanvasNode(new JMonetCanvas(new Dimension(5000, 5000)));
        pane.getChildren().add(myCanvasNode);
        Scene scene = new Scene(pane, 640, 480);
        stage.setScene(scene);
        stage.show();

        System.out.println("Memory in use after creating very large canvas: " + memoryInUseMb());

        Platform.runLater(() -> {
            
            // Dispose the canvas...
            myCanvasNode.dispose();
            myCanvasNode = null;
            
            // ... and remove it from the scene, and eliminate any reference to the parent Pane
            stage.setScene(new Scene(new Pane()));
            pane = null;
        });

        // May take a few GC cycles before everything is clean...
        Executors.newSingleThreadScheduledExecutor().scheduleAtFixedRate(new Runnable() {
            @Override
            public void run() {
                System.gc();
                System.runFinalization();
                System.out.println("Memory in use: " + memoryInUseMb());
            }
        }, 1, 1, TimeUnit.SECONDS);
    }
}
```

This should produce output similar to:

```
Initial memory in use: 21
Memory in use after creating very large canvas: 426
Memory in use: 296
Memory in use: 292
Memory in use: 4
```

**Comment by Javier-Pazos** [comment]
Thank you very much, Matt.
